app/models/ZuoraCatalog/ZuoraCatalog.scala

```diff
@@ -41,6 +41,10 @@ case class ZuoraCatalogPricePlan(
 )
 
 object ZuoraCatalogPricePlan {
+
+  def round(price: Double, precision: Int): Double =
+    BigDecimal(price).setScale(2, BigDecimal.RoundingMode.HALF_DOWN).toDouble
```

Comment:
You should use the `precision` value you're passing into the function here or remove the precision val if you don't need it to be configurable.